src/mon/Monitor.cc

```diff
@@ -674,31 +674,10 @@ void Monitor::handle_conf_change(const ConfigProxy& conf,
 
 void Monitor::update_log_clients()
 {
-  map<string,string> log_to_monitors;
-  map<string,string> log_to_syslog;
-  map<string,string> log_channel;
-  map<string,string> log_prio;
-  map<string,string> log_to_graylog;
-  map<string,string> log_to_graylog_host;
-  map<string,string> log_to_graylog_port;
-  uuid_d fsid;
-  string host;
-
-  if (parse_log_client_options(g_ceph_context, log_to_monitors, log_to_syslog,
-			       log_channel, log_prio, log_to_graylog,
-			       log_to_graylog_host, log_to_graylog_port,
-			       fsid, host))
-    return;
-
-  clog->update_config(log_to_monitors, log_to_syslog,
-		      log_channel, log_prio, log_to_graylog,
-		      log_to_graylog_host, log_to_graylog_port,
-		      fsid, host);
-
-  audit_clog->update_config(log_to_monitors, log_to_syslog,
-			    log_channel, log_prio, log_to_graylog,
-			    log_to_graylog_host, log_to_graylog_port,
-			    fsid, host);
+  auto parsed_options = parse_log_client_options(g_ceph_context, clog->get_log_channel());
+  clog->update_config(parsed_options);
```

Comment:
These two lines are often repeated. Would it potentially be worth implementing a wrapper function for these two commands? Or do you prefer the semantic separation of them?

Reply:
I actually had first coded the two as a single operation. But there were some log messages in the original code that
required access to the 'pasred_options' data. As I didn't wish to change the behavior even in the slightest, I've decided to keep the two lines as two separate calls.

Reply:
Makes sense!